### PR TITLE
Two issues: old associations missing in test, crash when old associations present in console.

### PR DESCRIPTION
**Pull request by allavena**
For me, there are two distinct issues. In test mode, behaviour is different than from the console. In test, the old version no longer has the associations - test app shows this. But the crash, I only get in console. I don't know how to reproduce in test mode (I think it would crash there too, if the associations were not lost). Here is the code reproducing the issue in console:

From the console:  (bundle exec bin/rails console)
    ActiveRecord::Migrator.migrate "db/migrate"
    d = Document.create_sample_document
    d_v2 = d.versions.last.reify(:has_many => true)
last line fails with 

NoMethodError: undefined method `section_id' for #<Section:0x00000001566318>
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/activemodel-4.2.3/lib/active_model/attribute_methods.rb:433:in`method_missing'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/paper_trail-4.0.0/lib/paper_trail/version_concern.rb:394:in `block (2 levels) in reify_has_many_through'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/activerecord-4.2.3/lib/active_record/relation/delegation.rb:46:in`map'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/activerecord-4.2.3/lib/active_record/relation/delegation.rb:46:in `map'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/paper_trail-4.0.0/lib/paper_trail/version_concern.rb:394:in`block in reify_has_many_through'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/paper_trail-4.0.0/lib/paper_trail/version_concern.rb:391:in `each'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/paper_trail-4.0.0/lib/paper_trail/version_concern.rb:391:in`reify_has_many_through'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/paper_trail-4.0.0/lib/paper_trail/version_concern.rb:347:in `reify_has_manys'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/paper_trail-4.0.0/lib/paper_trail/version_concern.rb:232:in`block in reify'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/paper_trail-4.0.0/lib/paper_trail/version_concern.rb:303:in `call'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/paper_trail-4.0.0/lib/paper_trail/version_concern.rb:303:in`without_identity_map'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/paper_trail-4.0.0/lib/paper_trail/version_concern.rb:168:in `reify'
    from (irb):3
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/railties-4.2.3/lib/rails/commands/console.rb:110:in`start'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/railties-4.2.3/lib/rails/commands/console.rb:9:in `start'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/railties-4.2.3/lib/rails/commands/commands_tasks.rb:68:in`console'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/railties-4.2.3/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
    from /home/andre/.rvm/gems/ruby-2.1.2/gems/railties-4.2.3/lib/rails/commands.rb:17:in`<top (required)>'

### CODE REVIEW

File: spec/model/document_spec.rb [hunk 1]
```diff
@@ -4,24 +4,17 @@
...
+    expect(d.title).to eq("d v2")
+    expect(d.sections.length).to eq(2)
+    expect(d.paragraphs.length).to eq(4)
+
     puts 'Previous version of document'
     d_v2 = d.versions.last.reify(:has_many => true)
     expect(d_v2.title).to eq("d v1")
-    expect(d.sections.length).to eq(2)
-    expect(d.paragraphs.length).to eq(4)
+    # The below fails. Associations are missing
```
Comment (jaredbeck) [inline spec/model/document_spec.rb:19]: The first version of the document did not have any associated records, right? `sections.length` _should_ return 0.  So, I don't think this demonstrates an issue in PaperTrail.

**Comment by allavena** [comment]
I respectfully disagree. The first version did have some associated records. I've updated the test case to make it clearer. 

The other pull request has a simpler setup and just shows the associations disappear during the tests (though they are present when running through the console)

PS: how is working in Ithaca? I did my PhD at Cornell years ago.

**Comment by jaredbeck** [comment]
> PS: how is working in Ithaca? I did my PhD at Cornell years ago. 

Small world! What was your field of study? Send me an email (jared@jaredbeck.com) and I'd be happy to chat.

**Comment by jaredbeck** [comment]
I'm going to close this, as I believe I've incorporated your changes into master with https://github.com/jaredbeck/pt_issue_590/commit/de54892426966d146a13c8c2382735826aa76660

Thanks!

**Pull request closed without merging**